parser.go

```diff
@@ -445,34 +446,80 @@ var (
 	nullLiteral  = []byte("null")
 )
 
+func createInsertComponent(keys []string, setValue []byte) []byte {
+	var buffer bytes.Buffer
+	buffer.WriteString(",\"")
+	buffer.WriteString(keys[0])
+	buffer.WriteString("\":")
+	for i := 1; i < len(keys); i++ {
+		buffer.WriteString("{\"")
+		buffer.WriteString(keys[i])
+		buffer.WriteString("\":")
+	}
+	buffer.Write(setValue)
+	buffer.WriteString(strings.Repeat("}", len(keys)-1))
+	return buffer.Bytes()
+}
+
 /*
-Get - Receives data structure, and key path to extract value from.
+
+Set - Receives existing data structure, path to set, and data to set at that key.
 
 Returns:
-`value` - Pointer to original data structure containing key value, or just empty slice if nothing found or error
-`dataType` -    Can be: `NotExist`, `String`, `Number`, `Object`, `Array`, `Boolean` or `Null`
-`offset` - Offset from provided data structure where key value ends. Used mostly internally, for example for `ArrayEach` helper.
-`err` - If key not found or any other parsing issue it should return error. If key not found it also sets `dataType` to `NotExist`
+`value` - modified byte array
+`err` - On any parsing error
 
-Accept multiple keys to specify path to JSON value (in case of quering nested structures).
-If no keys provided it will try to extract closest JSON value (simple ones or object/array), useful for reading streams or arrays, see `ArrayEach` implementation.
 */
-func Get(data []byte, keys ...string) (value []byte, dataType ValueType, offset int, err error) {
-	if len(keys) > 0 {
-		if offset = searchKeys(data, keys...); offset == -1 {
-			return []byte{}, NotExist, -1, KeyPathNotFoundError
+func Set(data []byte, setValue []byte, keys ...string) (value []byte, err error) {
+	// ensure keys are set
+	if len(keys) == 0 {
+		return nil, KeyPathNotFoundError
+	}
+
+	_, _, startOffset, endOffset, err := internalGet(data, keys...)
+	if err != nil {
+		if err != KeyPathNotFoundError {
+			// problem parsing the data
+			return []byte{}, err
```

Comment:
I see this in the Get method too (and likely you are following the existing practice), but what is the reason for initializing a new empty byte slice  as suppose to return nil? If an error occurs, it is more idiomatic to return nil so it is clearer in the code.